Remotion/ObjectBinding/Core/BusinessObjectPropertyPaths/BusinessObjectPropertyPathBase.cs

```diff
@@ -106,15 +106,21 @@ public override string ToString ()
               string.Format ("Property #{0} of property path '{1}' is not a single-value property.", propertyIndex, Identifier));
         }
 
-        var list = (IList) currentObject.GetProperty (currentProperty);
+        var list = (IList?) currentObject.GetProperty (currentProperty);
+        Assertion.DebugIsNotNull (
```

Comment:
Puh, I think that's actually a null bug and we should support list property returning null values.